apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/ClaimTransformerInterfaceV2.java

```diff
@@ -0,0 +1,17 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import java.util.Optional;
+import org.hl7.fhir.r4.model.ExplanationOfBenefit;
+
+/** Interface defining a transform method for claim processing. */
+@FunctionalInterface
+public interface ClaimTransformerInterfaceV2 {
```

Comment:
So here is the conundrum; definitely want to treat transformers via a std interface. So yes, only 3 (of 8) transformers actually care about the tax number, but how to create an interface / support two tracks. My original thoughts were:
1.  support two transform methods; one with and one without the includeTaxNumber. Doing that solves the case where 5 claim types don't need / care about that param, but comes as the cost of 5 transformers having to implement a method (entity, optional tax number) that will never be called.
2. have the transform interface support varargs; only need one method in implementation classes, but comes at the cost of having 5 transformers support a noop method.
3. have all transformers derive from a base class that implements the common interface. the base class could implement the 2 transform methods as basically doing nothing, and then a concrete class can override whichever transform is appropriate.

Reply:
One other question here...am going to change the interface to support 2 transform methods, one with the boolean and one without.
- public ExplanationOfBenefit transform(Object claimEntity);
- public ExplanationOfBenefit transform(Object claimEntity, Boolean includeTaxNumbers);

So for those claim types that have to have the includeTaxNumber, I plan on throwing an exception if the caller invokes the method that does not support the includeTaxNumber; that is pretty straight-forward.
Now for those claim types that don't care about the includeTaxNumber (and therefore call the appropriate method, without that param) what do you think I should do if they do call the method with includeTaxNumber? I can also throw and exception, or just handle things (ignoring the includeTax parameter). Thoughts???

Reply:
I am not a fan of interfaces that contain methods which will automatically throw if called on some implementations.  I have the exact same issue with some standard java interfaces like List and Set. 

I'd prefer to either have separate interfaces if it's truly vital that includeTaxNumbers not be provided to certain implementations or else broaden the definition of the method that takes the parameter to reflect that it can be ignored by some implementations.

Reply:
Not sure why this was resolved.  If these two methods are mutually exclusive then this needs to be two different interfaces.

Reply:
Alternatively a boolean property getter could be added to the interface to allow a caller to determine which method to call. Right now the call can blow up because the caller did not apply knowledge external to the interface about which method is appropriate to use. I prefer the 2 interfaces with instanceof test though.

Reply:
Based on our discussion adding a boolean method to the interface will allow the caller to know from the transformer itself which method is safe to call.  Since implementations fall into disjoint sets it might pay to have two abstract classes that the implementations can extend to eliminate the identical code in all of the implementing classes.